### PR TITLE
Convert documentation to org

### PR DESCRIPTION
**Pull request by person808**
- [x] Convert CONTRIBUTE.md

- [x] Convert CONVENTIONS.md

- [x] Convert DOCUMENTATION.md

- [x] Convert HOWTOS.md

- [x] Update helm-spacemacs filenames

- [x] Update link to documentation in home buffer

### CODE REVIEW

File: doc/CONVENTIONS.org [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+*** Reserved prefix
+
+**** User prefix
+
+~SPC o~ must not be used by any layer. It is reserved for the user.
+
+**** Major mode prefix
+
+~SPC m~ is reserved for the current major mode. Three keys bindings are
+not an issue (ie. ~SPC m h d~) since ~SPC m~ can be accessed via ~,~.
```
Comment (person808) [inline doc/CONVENTIONS.org]: The comma doesn't render properly. Bug?

Comment (tuhdo) [inline doc/CONVENTIONS.org]: It can, but with some Org settings. @CestDiego

Comment (CestDiego) [inline doc/CONVENTIONS.org]: Fixed in #1940

Comment (CestDiego) [inline doc/CONVENTIONS.org]: Problem still persists in Github Rendering though

**Comment by person808** [comment]
I'm using ¦ instead of pipe in some places because the regular pipe interferes with tables.

http://stackoverflow.com/questions/5144862/escape-pipe-character-in-org-mode

**Comment by person808** [comment]
DOCUMENTATION.org definitely needs to be reviewed thoroughly. Its so long I probably have made several errors somewhere.

**Comment by CestDiego** [comment]
question: do you edit them by hand? or do you use iedit?

**Comment by person808** [comment]
Hand, I've never tried iedit yet

**Comment by xfq** [comment]
Where is the `Spacemacs UI` section in `DOCUMENTATION.org`?

**Comment by xfq** [comment]
In https://github.com/person808/spacemacs/blob/d866423d3dfc2620507fc4bac822a8699178ab4a/doc/DOCUMENTATION.org#buffers-manipulation-key-bindings , `SPC b k` should be "kill a buffer", not "kill the current buffer".

**Comment by xfq** [comment]
In https://github.com/person808/spacemacs/blob/d866423d3dfc2620507fc4bac822a8699178ab4a/doc/DOCUMENTATION.org#ido-micro-state , "while in a `ido` buffer" should be "while in an `ido` buffer".

**Comment by syl20bnr** [comment]
Indeed I was merging this but then realized that 500 lines were missing :-D

**Comment by person808** [comment]
@xfq Wow, thanks for spotting that! I was wondering why the line count was so off.

**Comment by person808** [comment]
@syl20bnr I believe I have fixed the missing sections. Should be ready to merge now.

**Comment by xfq** [comment]
* https://github.com/person808/spacemacs/blob/docs/doc/DOCUMENTATION.org#buffers-manipulation-key-bindings
  * `SPC b k`: "kill the a buffer" -> "kill a buffer"

**Comment by person808** [comment]
@xfq Thanks again! Fixed

**Comment by syl20bnr** [comment]
Excellent :+1: 
This is it we are full .org now I believe :-)

Thank you !
Cherry-picked and squashed into `develop` branch, you can safely delete your branch.

**Comment by CestDiego** [comment]
Awesum! :)

**Pull request closed without merging**